clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -95,23 +92,11 @@
 
     private final Time time;
 
-    private final Map<Integer, Collection<ApiVersionsResponse.ApiVersion>> nodeApiVersions = new HashMap<>();
+    private final Map<Integer, NodeApiVersions> nodeApiVersions = new HashMap<>();
 
     private final Set<String> nodesNeedingApiVersionsFetch = new HashSet<>();
 
-    public NetworkClient(Selectable selector,
-                         Metadata metadata,
-                         String clientId,
-                         int maxInFlightRequestsPerConnection,
-                         long reconnectBackoffMs,
-                         int socketSendBuffer,
-                         int socketReceiveBuffer,
-                         int requestTimeoutMs,
-                         Time time,
-                         Collection<ApiVersionsResponse.ApiVersion> requiredApiVersions) {
-        this(NO_METADATA_UPDATER, metadata, selector, clientId, maxInFlightRequestsPerConnection,
-                reconnectBackoffMs, socketSendBuffer, socketReceiveBuffer, requestTimeoutMs, time, requiredApiVersions);
-    }
+    private final List<ClientResponse> abortedSends = new LinkedList<>();
```

Comment:
Oops, sorry, I meant to reply but I closed the browser window.  ArrayList seems like overkill here since we don't expect a lot of aborted sends to happen.  Also, the memory consumed by ArrayList only grows and never shrinks (unless you call trimToSize), which seems like a bad choice here.